### PR TITLE
Lock floating message 2

### PR DESCRIPTION
**Pull request by tomolimo**
Added a new style class .objectlockmessage
Added a button to force unlock when rights are in current profile
Tested in IE11, Edge 44.18362.387.0, Chrome 77.0.3865.90
Tests for FireFox don't pass: as the latest FF release 69.0.1 has a bug on the XHR async calls in beforeunload event.
Did some code cleansing.
refs #6322

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 

image

When rights are in the current profile, added a button to permit force unlock:
image

### CODE REVIEW

File: inc/objectlock.class.php [hunk 1]
```diff
@@ -151,30 +148,58 @@ function callUnlock( ) {
...
+   }
 
-      echo $ret;
+   /**
+    * Summary of getForceUnlockMessage
+    * @return string '' if no rights to unlock type,
+    *                else a <td></td> with a button to force unlock
+    */
+   private function getForceUnlockMessage() {
+      global $CFG_GLPI;
```
Comment (trasher) [inline inc/objectlock.class.php]: seems unused in metod

File: inc/objectlock.class.php [hunk 2]
```diff
@@ -151,30 +148,58 @@ function callUnlock( ) {
...
+      return $msg;
+   }
+
+
+   /**
+    * Summary of setLockedByYouMessage
+    * Shows 'Locked by You!' message and proposes to unlock it
+   **/
+   private function setLockedByYouMessage() {
+      global $CFG_GLPI;
```
Comment (trasher) [inline inc/objectlock.class.php]: seems unused in method

Comment (tomolimo) [reply inc/objectlock.class.php]: deleted

File: inc/objectlock.class.php (hunk 1 above)
Comment (tomolimo) [reply inc/objectlock.class.php]: deleted

File: inc/objectlock.class.php [hunk 3]
```diff
@@ -250,17 +266,22 @@ function askUnlock() {
...
+
       echo $ret;
 
-      $msg = "<table><tr><td class=red nowrap>";
+      $msg = "<span class='red' style='white-space: nowrap;'>";
 
-      $msg .= __('Locked by ')."<a href='".$user->getLinkURL()."'>$completeUserName</a> -> ".
-               Html::convDateTime($this->fields['date_mod']);
-      $msg .= "</td><td nowrap>";
+      $msg .= __('Locked by ')."<a href='".$user->getLinkURL()."'>".$userdata['name']."</a>";
```
[the remark this answers is not in the document]
Comment (tomolimo) [reply inc/objectlock.class.php]: One must use sprintf and not printf here:
`$msg .= sprintf(__('Locked by %s'), "<a href='" . $user->getLinkURL() . "'>" . $userdata['name'] . "</a>");`

**Comment by trasher** [comment]
ping?

**Pull request closed without merging**